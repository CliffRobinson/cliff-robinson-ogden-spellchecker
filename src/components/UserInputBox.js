
export function UserInputBox({userInputString, setUserInputString}) {

    const handleChange = (e) => {
        setUserInputString(e.target.value)
    }

    return (
    <input 
        type='text' 
        value={userInputString}
        onChange={handleChange}
    ></input>)
}

//CFR Comment 5: Now the user can input text to be parsed, and our component is now nice and functional.
// I normally like to write unit tests to confirm that the onchange is actually triggered, 
// but it's a little finicky, so for now on to processing the input string. 