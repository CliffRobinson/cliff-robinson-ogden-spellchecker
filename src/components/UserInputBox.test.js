import {shallow, mount} from 'enzyme'

import {UserInputBox} from './UserInputBox'

const quote = "Luke, I am your father."

test('userinputbox render', () => {
  const tree = mount(<UserInputBox userInputString={quote}/>)
  expect(tree).toMatchSnapshot()
})

//CFR Comment 4: Ok, we now have a test for the userInputBox, 
//and it only took a brief detour into Enzyme setup to get the snaps working properly!
//Let's get the box talking correctly to state now.