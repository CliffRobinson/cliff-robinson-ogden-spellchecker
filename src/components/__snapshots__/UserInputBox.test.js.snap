// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userinputbox render 1`] = `
<UserInputBox
  userInputString="Luke, I am your father."
>
  <input
    onChange={[Function]}
    type="text"
    value="Luke, I am your father."
  />
</UserInputBox>
`;
